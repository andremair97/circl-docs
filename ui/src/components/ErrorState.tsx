interface ErrorStateProps {
  title?: string;
  description?: string;
  onRetry?: () => void;
}

// ErrorState shows an error message and optional retry button.
export default function ErrorState({
  title = 'Something went wrong',
  description = 'Please try again.',
  onRetry,
}: ErrorStateProps) {
  return (
    <div className="py-8 text-center" data-testid="error-state">
      <h2 className="text-lg font-semibold text-bad">{title}</h2>
      {description && <p className="mt-2 text-sm text-gray-600">{description}</p>}
      {onRetry && (
        <button
          type="button"
          onClick={onRetry}
          className="mt-4 rounded border border-soft-border px-4 py-2 text-primary hover:bg-soft-border"
        >
          Retry
        </button>
      )}
    </div>
  );
}
