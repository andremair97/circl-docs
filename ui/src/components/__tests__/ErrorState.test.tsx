import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { vi } from 'vitest';
import ErrorState from '../ErrorState';

// Snapshot and behavior test for ErrorState component.
it('invokes retry handler when provided', async () => {
  const retry = vi.fn();
  const { container } = render(
    <ErrorState title="oops" description="fail" onRetry={retry} />
  );
  const user = userEvent.setup();
  await user.click(screen.getByRole('button', { name: /retry/i }));
  expect(retry).toHaveBeenCalled();
  expect(container).toMatchSnapshot();
});
