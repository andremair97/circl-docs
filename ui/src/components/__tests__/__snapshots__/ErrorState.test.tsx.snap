// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`invokes retry handler when provided 1`] = `
<div>
  <div
    class="py-8 text-center"
    data-testid="error-state"
  >
    <h2
      class="text-lg font-semibold text-bad"
    >
      oops
    </h2>
    <p
      class="mt-2 text-sm text-gray-600"
    >
      fail
    </p>
    <button
      class="mt-4 rounded border border-soft-border px-4 py-2 text-primary hover:bg-soft-border"
      type="button"
    >
      Retry
    </button>
  </div>
</div>
`;
